deduction_rules.py

```diff
@@ -265,9 +265,17 @@ def search_and_ban_in_subset(cells_to_check):
 
 
 def yswing(sudoku):
-    '''RULE: if (i,j)=AC and (k,l)=BC and (i,l)=AB then remove C from (j,k).'''
+    '''RULE: If three of the corners of a rectangular have two candidates AC, AB and BC, respectively, then C can be removed from the fourth corner.
+    Note that we don't actually need a rectangular, it is also enought to have a circle.
+    E.g. (C+ means that there can be other numbers as C)
+     ============================
+     || C+  AB  C || -  AC  -  ||
+     || -   -   - || -  -   -  ||
+     || BC  -   - || C+ C+  C+ ||
+     ============================'''
     # (i,j)=AC and (k,l)=BC
     def allowed_nums_multicells(*args):
+        """Input: cells. Output: numbers that can be written in all cell."""
         return set.intersection(*[set(_allowed_numbers(sudoku,[arg])[0]) for arg in args])
```

Comment:
The `set` conversion in the list comprehension can be removed, as `set.intersect` accepts all types of `Iterable`s as its arguments.